fiftyone/brain/internal/core/leaky_splits.py

```diff
@@ -0,0 +1,421 @@
+            )
+
+    def set_threshold(self, threshold):
+        """Set threshold for leak computation"""
+        self._leak_threshold = threshold
+
+    @property
+    def leaks(self):
+        """
+        Returns view with all potential leaks.
+        """
+
+        # access cache if possible
+        if self._last_computed_threshold == self._leak_threshold:
+            return self._leaks
+
+        # populate index if it doesn't have some samples
+        if not self._similarity_index.total_index_size == len(self.samples):
+            (
+                embeddings,
+                sample_ids,
+                label_ids,
+            ) = self._similarity_index.compute_embeddings(self.samples)
+            self._similarity_index.add_to_index(
+                embeddings, sample_ids, label_ids
+            )
+
+        # check if duplicates already computed in the index
+        if not self._similarity_index.thresh == self._leak_threshold:
+            self._similarity_index.find_duplicates(self._leak_threshold)
+        duplicates = self._similarity_index.duplicates_view()
+
+        # filter duplicates to just those with neighbors in different splits
+        to_keep = []
+        for (
+            sample_id,
+            neighbors,
+        ) in self._similarity_index.neighbors_map.items():
+            keep_sample = False
+            sample_split = self._id2split.get(sample_id, None)
+            if sample_split is None:
+                _throw_index_is_bigger_warning(sample_id)
+                continue  # sample is in index but not passed to leaky_splits
+            leaks = []
+            for n in neighbors:
+                neighbor_split = self._id2split.get(n[0], None)
+                if neighbor_split is None:
+                    _throw_index_is_bigger_warning(n[0])
+                    continue
+                if not (neighbor_split == sample_split):
+                    # at least one of the neighbors is from a different split
+                    # we keep this one
+                    keep_sample = True
+                    leaks.append(n[0])
+
+            if keep_sample:
+                to_keep.append(sample_id)
+                # remove all other samples because they are all from the same split
+                to_keep = to_keep + leaks
+
+        duplicates = duplicates.select(to_keep)
+
+        # cache to avoid recomputation
+        self._last_computed_threshold = self._leak_threshold
+        self._leaks = duplicates
+
+        return duplicates
+
+    @property
+    def num_leaks(self):
+        """Returns the number of leaks found."""
+        return self.leaks.count
+
+    def leaks_for_sample(self, sample):
+        """Return view with all leaks related to a certain sample.
+
+        Args:
+            sample: sample object or sample id
+
+        """
+        # compute leaks if it hasn't happend yet
+        if not self._leak_threshold == self._last_computed_threshold:
+            leaks = self.leaks
```

Comment:
Is this variable used? Looks like it isn't

Reply:
it can be an underscore instead. Really I'm just trying to invoke the self.leaks property so that the computation occurs.

Reply:
made the change